zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -183,6 +190,335 @@ pub(crate) fn get_legacy_transparent_address<P: consensus::Parameters>(
     Ok(None)
 }
 
+/// Returns the transparent address index at the start of the first gap of at least `gap_limit`
+/// indices in the given account, considering only addresses derived for the specified key scope.
+///
+/// Returns `Ok(None)` if the gap would start at an index greater than the maximum valid
+/// non-hardened transparent child index.
+pub(crate) fn find_gap_start(
+    conn: &rusqlite::Connection,
+    account_id: AccountRef,
+    key_scope: KeyScope,
+    gap_limit: u32,
+) -> Result<Option<NonHardenedChildIndex>, SqliteClientError> {
+    match conn
+        .query_row(
+            r#"
+            WITH offsets AS (
+                SELECT
+                    a.transparent_child_index,
+                    LEAD(a.transparent_child_index)
```

Comment:
Check what SQLite version introduced this syntax.

Reply:
This was introduced in 3.25.0